### ISSUE TITLE
Sprint Feb 19 to March 1, 2024

### ISSUE THREAD

**Issue by mariusandra** [PostHog/posthog#20328]
# Global Sprint Planning

## 3 things that might take us down

1. Ted running a PG migration (posted in #dev)
2. DNS - Terraform all of cloudflare -- already on the list
3. 

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by mariusandra** [comment]
# Team Product Analytics

**Support hero:** 
- Tom

## Retro

- 👍 HogQL rewrite
  - ✅ Lifecycle - out
  - ✅ Retention - out
  - ✅ Stickiness - out (**this is new**)
  - 🔲 Trends - being thoroughly tested, hope to wrap up this week
  - 🟦 Funnels - started, strong progress
  - 🟦 Paths - waiting for funnels for last bit
- 😍 HogQL Autocomplete

## OKR Q1 2024

1. HogQL & Data Exploration (Julian, Marius)
   - Convert ALL our insights to use HogQL as their base. **ongoing**
   - Improve the type system and get rid of `assumeNotNull`.
   - Write great docs for HogQL and data exploration nodes. **ongoing**
   - Proactively monitor query performance.

2. Product Analytics frontend (Thomas, Michael)
   - Clean things up now that we have PostHog 3000 and HogQL everywhere. **ongoing**
   - Work on the universal "Explore" view. **ongoing**

3. BI (Tom)
   - Support non-event data sources on HogQL insights.
   - Build the new querying experience, and/or integrate it into the "Explore" view.

<details>
<summary>Action plan</summary>
- Come January, we'll take [this list](https://github.com/PostHog/meta/issues/130), split the checkboxes between the team, and keep going till it's done. ETA: end of January.
- In February (leap year, woohoo), we'll make Product Analytics ~great~ spark joy again: insights UI cleanup, unified data exploration experience, adding missing analysis features (lifecycle breakdowns, etc), and in parallel work on the BI and the HogQL editing experience. What _exactly_ we'll do and in what order is still TBD.
- In March we'll keep going. The goal is to have an amazing BI / Product Analytics / Insights / Explore / HogQL experience by the end of it.
</details>

## Next sprint

### High priority

Rough first draft. To be discussed at standup.

- HogQL insights rewrite last steps
  - paths, funnels - @thmsobrmlr
  - trends - @mariusandra  

TBD


### Low priority / side quests

- Django upgrade - @webjunkie

**Comment by robbie-c** [comment]
# Team Web Analytics

Last sprint 
TODO

## Hang over items from previous sprint
TODO

## OKR
1. Finish the [beta to-do list](https://github.com/PostHog/posthog/issues/18547)
2. Find 5 high-ICP reference customers

### High priority
* Performance
* Add ability to separate sites (e.g. Sites scene or saved filters)

### Low priority
* polish and be responsive to support

**Comment by pauldambra** [comment]
# Team Replay-i

**Support hero:** David

## Retro

- stuff carried on without me which is a good signal @pauldambra 
- blob incidents worrying when on call @marandaneto 
- CH knowledge still not widely spread @daibhin 

## Hang over items from previous sprint

- 🟡 RN recordings - we know what we need to do after spiking
- 🟡 using embeddings/clustering to generate session summaries @daibhin @pauldambra
- ✅ we startedd iOS recording so we can pull from larger tester pool @marandaneto
- 🟡 2 more tester/user for android replay - Inarix started Android testers - lots of good feedback
- 🟡 do we release comments/discussion @daibhin
   - 15ish people enabled it but not really using it
   - we want to follow-up with folk that turned this on in feature preview
- ✅ buffering/loading/performance @pauldambra

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red

- 🟡📱 happy android replay users
   - feedback from beta testers but not a working recording from them yet 😓
- 🟡 people love replay even more
   - still more bug reports than we'd like
- 🟡 first class mobile support
   - not focussed here, focus is on replay
   - but making small pieces of progress

### High priority

- iOS session replay have first captured @marandaneto 
- 5 Open Source Android apps tested with android replay @pauldambra 
- get "show similar recordings" out for our team  @daibhin @pauldambra 
- start generating clusters from recording embeddings for our team @daibhin @pauldambra 

### Low priority / side quests

- contribute the network capture back to rrweb - so they don't make one that doesn't work for us - @pauldambra
- network payload capture on onboarding? @pauldambra 
- adding billing limits to unsubsscribe? @pauldambra 
- how much embedding/clustering can we do just with session recording metadata (when can we avoid Open AI) @daibhin 
- incre

**Comment by benjackwhite** [comment]
# Team Infra

**Support hero:** @frankh 

## Hang over items from previous sprint

1. 🟡 Keep rolling out the new ingestion ingresses with client-side support @frankh 
  - Lots of small edge case issues still taking time to investigate
  - JS is ready to go with new endpoints - just needs testing it
  - Goal is to do that rollout and then _only_ use Cloudflare for assets (direct to NLB otherwise)
2. 🟡 Move the final work into `charts` (posthog in particular) @danielxnj 
  - Almost there but caused an incident 😱 due to unforeseen way that helmfile works
3. ✅ Zero downtime kafka rollover for EU


## OKR

1. 💪 Deploy with confidence ✅
1. 💰 Keep costs under control ✅
1. 👀 More Observability 🟡
1. 🎯 Soc2 ✅

### High priority

1. Finalise `charts` movement and get Argo CD canary deploys working @danielxnj 
- Make web the focus (capture as next target)
2. Rollout all clients to new capture endpoints @frankh 
- And keep on top of remaining edge cases

### Low priority / side quests

**Comment by benjackwhite** [comment]
# Team CDP


### High priority

- Finish the CDP RFC roadmap
- Support Robbie with scaling Web Analytics 
- Sites or Saved Filters for Web Analytics (influences CDP sources work)
- PoC out the alternative $person ingestion work

**Comment by neilkakkar** [comment]
# Team Feature Success

**Support hero:** @liyiy (surveys) + @neilkakkar    
Days off:
Juraj: 2 days
Neil: 1 days
Li: 1 days

## Retro

- Experiments UI create & view overhaul - @jurajmajerik -> pushed into next sprint, taking longer to sort out designs 
- Fix control is never significant in experiments - @jurajmajerik -> done
- Survey results are squished fix; delayed surveys; nicer results loading screens; more results filtering in surveys - @liyiy
- Statistical significance on secondary metrics & display secondary metrics charts - @neilkakkar  -> done
- Refactor FeatureFlagReleaseConditions to work well with scheduled changes, surveys, and flags - @neilkakkar -> in progress, will roll over a day or two, higher priority to decide billing issues



## Hang over items from previous sprint

---

## OKRs

1. Doubling down on surveys
2. Doubling down on experiments

### High priority

- Experiments UI create & view overhaul - @jurajmajerik

- Survey results are squished fix; delayed surveys; nicer results loading screens; more results filtering in surveys - @liyiy
- Find a good flow for restricting users to exposed users in secondary metrics for experiments - @neilkakkar 
- Add property filters to event matching in cohorts - @neilkakkar 


### Low priority / side quests

- Temporal queues for feature success - @neilkakkar 

---

We'll do - Event based triggers for surveys (RFC here: https://github.com/PostHog/product-internal/pull/556/ )- @liyiy  - after this sprint.

**Comment by EDsCODE** [comment]
# Team Data Warehouse

### OKR Q1 2024

### Objective

**Get EU moved over to new cluster**

* **Key Results:**
    * Terraform, ansible, data migrated over, backups, housewatch
    * Deprecate old cluster

**ByConity evaluation**

* **Key Results:**
    * Load up with all clickhouse data and benchmark

**5 reference customers with qualities:**

* **Key Results:**
    * Paying a reasonable amount
    * Tell us they’re delighted
    * Not a big edge case
    * Using our in house data import pipeline
    * Weekly queries on a data warehouse connected table
    * Saving queried tables to insights?

**Data Warehouse queries are compatible with insights hogQL**

 

### Retro

**Product**
- [x] stop flaky jobs
- [x] test postgres integration live
- ~[ ] experiment with adding imported data to persons table~

**Infra**
- [ ] Debug incremental backing up of large tables
- [ ] TF configs for all EU CH nodes. Core, Coordinator, ZK
- [ ] Ansible configs for all EU CH nodes. Core, Coordinator, ZK

### High priority

**Product**
- [ ] @EDsCODE data warehouse data in trends
- [ ] @Gilbert09 View/table linking precursor to linking data warehouse data to person models

**Infra**

**Comment by raquelmsmith** [comment]
# Team Growth

## Retro
<details>
<summary>Retro items</summary>
Lots of things in progress / left over
### High priority
- [x] Make new celery queue for usage reports
- [ ] Full testing of the onboarding flow @raquelmsmith - partially done
- [x] Fix mrr per product for credit-based annual customers - @raquelmsmith https://github.com/PostHog/billing/pull/461
- [x] Feature flag to turn off quota limiting during billing migrations @xrdt 
  - Basically done https://github.com/PostHog/posthog/pull/19996
- [ ] Don't drop data for 7 (or maybe 3) days until after it's done - needs a relaunch
- [ ] Onboarding reskin based on Cory's mocks @xrdt - in progress
- [ ] Teams plan @raquelmsmith - in progress
- [ ] Come up with 5 ideas to bump the session replay activation rate @raquelmsmith 
- [ ] Rethink some activation metrics (remove team member invited, double check rates with diff activities) @raquelmsmith - in progress
- [ ] Overlap in billing gauge projected / current / billing limit lines. https://posthoghelp.zendesk.com/agent/tickets/10070 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/12692)
</details>


## OKR
- 🟡 Improve activation rates for product analytics and session replay
  - 🟢  Polish onboarding flow with Cory
  - 🟡 Run 1 experiment per week
  - 🟡 Do an average of 2 customer interviews per week
- ⚪ Launch pricing for Data warehouse, queries, CDP


## This sprint
### High priority
- [ ] Product intro experiment is doing poorly! @raquelmsmith 
- [ ] Full testing of the onboarding flow @raquelmsmith - partially done
- [ ] Don't drop data for 7 (or maybe 3) days until after it's done
    - [ ] Need to reimplement this, draft PR open here: https://github.com/PostHog/posthog/pull/20236
- [ ] Onboarding reskin based on Cory's mocks @xrdt - in progress
    - [ ] config screen
    - [ ] products screen
- [ ] Teams plan @raquelmsmith - in progress
- [ ] 21M events free plan @raquelmsmith - in progress
- [ ] Rethink some activation metrics (remove team member invited, double check rates with diff activities) @raquelmsmith - in progress
- [ ] Come up with 5 ideas to bump the session replay activation rate @raquelmsmith @xrdt 


### Low priority / side quests
- [ ] Write two articles for Andy
